shardingsphere-distribution/shardingsphere-proxy-distribution/src/main/resources/bin/start.sh

```diff
@@ -137,12 +156,57 @@ if [ -z "$PORT" ]; then
 fi
 
 CLASS_PATH=${CONF_PATH}:${CLASS_PATH}
-MAIN_CLASS=${MAIN_CLASS}" "${PORT}" "${CONF_PATH}" "${ADDRESSES}
+MAIN_CLASS="${MAIN_CLASS} ${PORT} ${CONF_PATH} ${ADDRESSES}"
+
+if [ $PORT = -1 ]; then
+  REAL_PORT=3307
+else
+  REAL_PORT=$PORT
+fi
 
-echo "Starting the $SERVER_NAME ..."
 echo "The classpath is ${CLASS_PATH}"
 echo "main class ${MAIN_CLASS}"
-
-nohup java ${JAVA_OPTS} ${JAVA_MEM_OPTS} -classpath ${CLASS_PATH} ${MAIN_CLASS} >> ${STDOUT_FILE} 2>&1 &
-sleep 1
-echo "Please check the STDOUT file: $STDOUT_FILE"
+echo "STDOUT log file: $STDOUT_FILE"
+echo -e "Starting the $SERVER_NAME ...\c"
+
+for((i=1;i<=10;i++)); do
+  if [ "$i" = "10" ]; then
+    echo "failed: Address already in use"
+    exit
+  fi
+
+  PORT_STATUS=$(netstat -ant |$GREP $REAL_PORT |$GREP LISTEN)
```

Comment:
ShardingSphere-Proxy could listen on specified address such as `192.168.0.1/red`.

Reply:
ok, i fix it